hudi-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataPayload.java

```diff
@@ -177,10 +179,12 @@ public HoodieMetadataPayload preCombine(HoodieMetadataPayload previousRecord) {
   /**
    * Returns the files added as part of this record.
    */
-  public FileStatus[] getFileStatuses(Path partitionPath) {
+  public FileStatus[] getFileStatuses(Configuration hadoopConf, Path partitionPath) throws IOException {
+    FileSystem fs = partitionPath.getFileSystem(hadoopConf);
+    long blockSize = fs.getDefaultBlockSize(partitionPath);
     return filterFileInfoEntries(false)
-        .map(e -> new FileStatus(e.getValue().getSize(), false, 0, 0, 0, 0, null, null, null,
-            new Path(partitionPath, e.getKey())))
+        .map(e -> new FileStatus(e.getValue().getSize(), false, 0, blockSize, 0, 0,
```

Comment:
@umehrot2 Since we are fixing 0 -> blockSize, just curious, are the other `0` arguments valid or should they also have some dynamic value ? We don't have to fix it as part of the PR but would be great if you can take a look..

Reply:
I don't know the entire flow. but wondering why not set every arg in new FileStatus(....) from e.